google/cloud/documentai_toolbox/wrappers/document_revision.py

```diff
@@ -0,0 +1,502 @@
+        if p.revision_id:
+            while p:
+                p = p.parent
+                level += 1
+        return level
+    except Exception:
+        return 0
+
+
+def _print_child_tree(
+    current_revision: "DocumentWithRevisions",
+    doc: "DocumentWithRevisions",
+    seen: List[str],
+):
+    r"""Prints the revision tree of a child revision.
+
+    Args:
+        None.
+
+    Returns:
+        None.
+    """
+    tab = "  " * _get_level(doc)
+
+    if doc.children:
+        if current_revision.revision_id == doc.revision_id:
+            print(tab + "└──>", end="")
+        else:
+            print(tab + "└──", end="")
+        print(doc.revision_id)
+        for each in doc.children:
+            seen.append(each.revision_id)
+            _print_child_tree(current_revision, each, seen)
+    else:
+        if current_revision.revision_id == doc.revision_id:
+            print(tab + "└──>", end="")
+        else:
+            print(tab + "└──", end="")
+        print(doc.revision_id)
+
+
+@dataclasses.dataclass
+class DocumentWithRevisions:
+    r"""Represents a wrapped Document.
+
+    A single Document protobuf message with revisions will written as several `dp.bp` files on
+    GCS by Document AI's Processor Training UI.  This class hides away the
+    shards from the users and implements convenient methods for reading and moving between different document revisions.
+
+    Users should initilize the DocumentWithRevision class by using `from_gcs_prefix_with_revisions` function.
+
+    Attributes:
+        document (documentai.Document):
+            Required. The Document AI document.
+        revision_nodes (List[documentai.Document]):
+            Required. A list of Document AI document revisions.
+        gcs_bucket_name (str):
+            Required. The gcs bucket.
+
+            Format: Given `gs://{bucket_name}/{optional_folder}/{operation_id}/` where `gcs_bucket_name={bucket_name}`.
+        gcs_prefix (str):
+            Required.The gcs path to a single processed document.
+
+            Format: `gs://{bucket}/{optional_folder}/{operation_id}/{folder_id}`
+                    where `{operation_id}` is the operation-id given from BatchProcessDocument
+                    and `{folder_id}` is the number corresponding to the target document.
+        parent_ids (List[str]):
+            Required. A list of parent ids.
+            Parents are defined as nodes with 1 or more children.
+        all_node_ids (List[str]):
+            Required. A list of all the node ids.
+    """
+
+    document: Document = dataclasses.field(repr=False)
+    revision_nodes: List[documentai.Document] = dataclasses.field(repr=False)
+    gcs_bucket_name: str = dataclasses.field(repr=False, default=None)
+    gcs_prefix: str = dataclasses.field(repr=False, default=None)
+    parent_ids: List[str] = dataclasses.field(repr=False, default=None)
+    all_node_ids: List[str] = dataclasses.field(repr=False, default=None)
```

Comment:
I would expect `parent_ids` and `all_node_ids` are things that could be extracted from the data on GCS in `__post_init__`? (as opposed to requiring them in `__init__`, that is)